common/lib/xmodule/xmodule/js/spec/problem/edit_spec_hint.coffee

```diff
@@ -0,0 +1,772 @@
+    </stringresponse>
+
+    </problem>
+    """)
+
+  it 'text input with or= with hints', ->
+    data = MarkdownEditingDescriptor.markdownToXml(""">>Where Paris?<<
+      = France		{{ BRAVO::hint1}}
+      or= USA			{{   meh::hint2  }}
+
+    """)
+    expect(data).toEqual("""
+    <problem>
+    <p>Where Paris?</p>
+    <stringresponse answer="France" type="ci" >
+      <correcthint label="BRAVO">hint1</correcthint>
+      <additional_answer answer="USA"><correcthint label="meh">hint2</correcthint></additional_answer>
+      <textline label="Where Paris?" size="20"/>
+    </stringresponse>
+    
+    </problem>
+    """)
+
+  it 'text input with demand hints', ->
+    data = MarkdownEditingDescriptor.markdownToXml(""">>Where Paris?<<
+          = France		{{ BRAVO::hint1 }}
+
+          || There are actually two countries with cities named Paris. ||
+          || Paris is the capital of one of those countries. ||
+
+    """)
+    expect(data).toEqual("""
+    <problem>
+    <p>Where Paris?</p>
+    <stringresponse answer="France" type="ci" >
+      <correcthint label="BRAVO">hint1</correcthint>
+      <textline label="Where Paris?" size="20"/>
+    </stringresponse>
+    
+    <demandhint>
+      <hint>There are actually two countries with cities named Paris.</hint>
+      <hint>Paris is the capital of one of those countries.</hint>
+    </demandhint>
+    </problem>""")
+
+
+describe 'numeric input components', ->
+  it 'numeric input with hints', ->
+    data = MarkdownEditingDescriptor.markdownToXml("""
+        >>Enter the numerical value of Pi:<<
+        = 3.14159 +- .02   {{ Pie for everyone!   }}
+
+        >>Enter the approximate value of 502*9:<<
+        = 4518 +- 15%  {{PIE:: No pie for you!}}
+
+        >>Enter the number of fingers on a human hand<<
+        = 5
+        
+
+    """)
+    expect(data).toEqual("""
+    <problem>
+    <p>Enter the numerical value of Pi:</p>
+    <numericalresponse answer="3.14159">
+      <responseparam type="tolerance" default=".02" />
+      <formulaequationinput label="Enter the numerical value of Pi:" />
+      <correcthint>Pie for everyone!</correcthint>
+    </numericalresponse>
+
+    <p>Enter the approximate value of 502*9:</p>
+    <numericalresponse answer="4518">
+      <responseparam type="tolerance" default="15%" />
+      <formulaequationinput label="Enter the approximate value of 502*9:" />
+      <correcthint label="PIE">No pie for you!</correcthint>
+    </numericalresponse>
+
+    <p>Enter the number of fingers on a human hand</p>
+    <numericalresponse answer="5">
+      <formulaequationinput label="Enter the number of fingers on a human hand" />
+    </numericalresponse>
+
+
+    </problem>
+    """)
+
+  # The output xml here shows some of the quirks of how historical markdown parsing does or does not put
+  # in blank lines.
+  # TODO nparlante: there's a bug in the parsing stack if the >>text<< is the same twice
```

Comment:
Hey, there's a TODO doing it's job. That's a bug I noticed when I was up to my eyeballs running through parsing cases. Anyway, now I just tried a zillion different ways and it works fine, so it must have been some transient issue in the code, or programmer fever dream.